src/field_theory/galois.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2020 Thomas Browning and Patrick Lutz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning and Patrick Lutz
+-/
+
+import field_theory.normal
+import field_theory.primitive_element
+import field_theory.fixed
+import ring_theory.power_basis
+
+/-!
+# Galois Extensions
+
+In this file we define Galois extensions as extensions which are both separable and normal.
+
+## Main definitions
+
+- `is_galois F E` where `E` is an extension of `F`
+- `fixed_field H` where `H : subgroup (E ≃ₐ[F] E)`
+- `fixing_subgroup K` where `K : intermediate_field F E`
+- `galois_correspondence` where `E/F` is finite dimensional and Galois
+
+## Main results
+
+- `fixing_subgroup_of_fixed_field` : If `E/F` is finite dimensional (but not necessarily Galois)
+  then `fixing_subgroup (fixed_field H) = H`
+- `fixed_field_of_fixing_subgroup`: If `E/F` is finite dimensional and Galois
+  then `fixed_field (fixing_subgroup K) = K`
+Together, these two result prove the Galois correspondence
+-/
+
+noncomputable theory
+open_locale classical
+
+open finite_dimensional alg_equiv
+
+
+section
+
+variables (F : Type*) [field F] (E : Type*) [field E] [algebra F E]
+
+/-- A field extension E/F is galois if it is both separable and normal -/
+@[class] def is_galois : Prop := is_separable F E ∧ normal F E
+
```

Comment:
Since we're keeping the `is_galois` typeclass, let's register its projections:
```suggestion

instance is_galois.to_is_separable [h : is_galois F E] : is_separable F E := h.1

instance is_galois.to_normal [h : is_galois F E] : normal F E := h.2
```